step-core/src/main/java/step/resources/ResourceManager.java

```diff
@@ -1,159 +1,158 @@
-		if (resourceType != null) {
-			criteria.put("resourceType", resourceType);
-		}
-		List<Resource> res = findManyByCriteria(criteria);
-		if (objectPredicate != null) {
-			res = res.stream().filter(objectPredicate).collect(Collectors.toList());
-		}
-		return res;
-	}
-
-	ResourceRevisionContentImpl getResourceRevisionContent(String resourceRevisionId) throws IOException;
-
-	ResourceRevision getResourceRevision(String resourceRevisionId);
-
-	String getResourcesRootPath();
-
-	Resource createTrackedResource(String resourceType,
-								   boolean isDirectory,
-								   InputStream resourceStream,
-								   String resourceFileName,
-								   ObjectEnricher objectEnricher,
-								   String trackingAttribute,
-								   String actorUser, String origin,
-								   Long originTimestamp) throws IOException, InvalidResourceFormatException;
-
-	/**
-	 * Create a copy of the resource from a source repository
-	 *
-	 * @param resource              the source resource {@link Resource} to be copied
-	 * @param sourceResourceManager the source resource manager to copy from
-	 * @param actorUser
-	 * @return the newly create resource {@link Resource}
-	 * @throws IOException, SimilarResourceExistingException or InvalidResourceFormatException that may occur during the call
-	 */
-	Resource copyResource(Resource resource, ResourceManager sourceResourceManager, String actorUser) throws IOException, InvalidResourceFormatException;
-
-	/**
-	 * Save the content provided as stream to an existing resource.
-	 * This creates a new {@link ResourceRevision} for the {@link Resource}
-	 * and saves the content provided as stream under this revision.
-	 *
-	 * @param resourceId       the id of the resource to be updated
-	 * @param resourceStream   the stream of the resource to be saved
-	 * @param resourceFileName the name of the resource (filename)
-	 * @param actorUser
-	 * @return the updated {@link Resource}
-	 * @throws IOException an IOException occurs during the call
-	 */
-	Resource saveResourceContent(String resourceId, InputStream resourceStream, String resourceFileName, String actorUser)
-			throws IOException, InvalidResourceFormatException;
-
-	/**
-	 * Saved the resource object only
-	 * @param resource the resource to be saved
-	 * @return the updated {@link Resource}
-	 * @throws IOException an IOException occurs during the call
-	 */
-	Resource saveResource(Resource resource) throws IOException;
-
-	ResourceRevision saveResourceRevision(ResourceRevision resourceRevision) throws IOException;
-
-	/**
-	 * Delete the resource and all its revisions 
-	 * 
-	 * @param resourceId the id of the {@link Resource} to be deleted
-	 */
-	void deleteResource(String resourceId);
-
-	void deleteResourceRevisionContent(String resourceId);
-
-	List<Resource> findManyByCriteria(Map<String, String> criteria);
-
-	default void cleanup() {
-	}
-
-}
+/*******************************************************************************
```

Comment:
(unrelated to this specific PR) - we really need to implement this "formatting standardization + reformat entire codebase" thing. This is the 3rd or 4th file in this PR which has been completely rewritten just because one or two lines changed, but the entire file was reformatted. Not blaming you, but IntelliJ ;)